presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionDefinitionProvider.java

```diff
@@ -27,12 +36,33 @@
     @Override
     public UdfFunctionSignatureMap getUdfDefinition(String filePath)
     {
+        final int maxDepth = 4;
```

Comment:
make this private final static at the top of the class. also we could consider making this configurable.